CRM/Member/Form/MembershipRenewal.php

```diff
@@ -339,6 +339,12 @@ public function buildQuickForm() {
 
     $this->applyFilter('__ALL__', 'trim');
 
+    // CRM-20571
+    // Get the Join Date from Membership info as it is not available in the Renew form
+    $joinDate = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $this->_id, 'join_date');
+    // Save its value in a hidden field
+    $this->addElement('hidden', 'join_date', $joinDate);
```

Comment:
Idt, there is a doc which explain "each parameter of each function in CiviCRM" as you mention. Just pinging @eileenmcnaughton if she might have seen any. What I found is https://docs.civicrm.org/dev/en/master/core/architecture/#form which explains this basic form function.